docs/reference/search/search.asciidoc

```diff
@@ -654,7 +654,8 @@ exclude fields from this subset using the `excludes` property.
 (Optional, array of strings)
 Stats groups to associate with the search. Each group maintains a statistics
 aggregation for its associated searches. You can retrieve these stats using the
-<<indices-stats,indices stats API>>.
+<<indices-stats,indices stats API>>. At maximum, only 1,000 groups will be
+tracked. Beyond this number, no new groups will be added to the stats.
```

Comment:
does this cause a silent failure? Should we consider failing the update settings call when the group number goes too high?

Reply:
Also, curious about the 1000 number. Did you see real cases where a user had more than 1000?

Reply:
We ignore groups past the 1000 mark. There is no failure, they will just not be tracked. I could tweak the wording here to make that clearer. I also considered an LRU strategy to bound the size. But this is tricky -- the search stats logic increments then decrements certain stats, which is based on the assumption that stats groups are never evicted.

Also, the setting just exists to make testing easier ("This setting exists only for test purposes, and cannot be changed outside of tests").

As for 1000, I just chose a large number that I think users would not reasonably hit. I did see a case where a user had more than 1000 groups, when they were adding a new group per user session ID. This was creating far too many stats groups, and they were able to find another way to solve their use case. I think we should definitely discuss the 1000 number and whether we think it could affect users.

Reply:
I am not too sure about the implications but I would consider failing hard rather than silently not tracking when groups are too many. I feel like it could be seen as a breaking change from some users, but I would also expect users to have noticed issues caused by too many groups, hence this can also be seen like a bugfix as too many groups hinder stability?